### DISCUSSION TITLE
Query raw error

### DISCUSSION THREAD

**Discussion by theart84** [prisma/prisma#17785] [Q&A]
Have a request:
```
return this.prismaService.$queryRaw(Prisma.sql`
          select count(*)::integer, date_part('minute', v.timestamp) as "timestamp"
          from "Visitor" v
          where (v.timestamp between ${currentFrom}::timestamp and  ${currentTo}::timestamp) and project_tenant_id = ${project_tenant_id}
          group by date_part('minute', v.timestamp)
          order by timestamp
      `);
```
This query works well.
I need to pass a dynamic value to the date_part function and I do this:
```
return this.prismaService.$queryRaw(Prisma.sql`
          select count(*)::integer, date_part(${date}, v.timestamp) as "timestamp"
          from "Visitor" v
          where (v.timestamp between ${currentFrom}::timestamp and  ${currentTo}::timestamp) and project_tenant_id = ${project_tenant_id}
          group by date_part(${date}, v.timestamp)
          order by timestamp
      `);
```
This request generates an error:
```
"Raw query failed. Code: `42803`. Message: `db error: ERROR: column \"v.timestamp\" must appear in the GROUP BY clause or be used in an aggregate function`",
```
What could be the problem?
Table:
<img width="1261" alt="image" src="https://user-images.githubusercontent.com/64737975/217258171-d1cbdffb-9cfd-40fc-9674-43ef17586098.png">

**Comment by nurul3101** [comment]
Hello @theart84 👋 
Which version of Prisma are you using?
Can you try passing the variable like this [example](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#using-variables)? Do you get the same error in this case as well?

**Reply by theart84** [reply]
HI!
"prisma": "4.9.0",
"@prisma/client": "4.9.0"
Method:
```
public async getMetricForVisitor( project_tenant_id: string; date: DateFormat; from: string; to: string) {
    return this.prismaService.$queryRaw(Prisma.sql`
          select count(*)::integer, date_part(${date}, v.timestamp) as "timestamp"
          from "Visitor" v
          where (v.timestamp between ${from}::timestamp and  ${to}::timestamp) and project_tenant_id = ${project_tenant_id}
          group by date_part(${date}, v.timestamp)
          order by timestamp
      `);
}
```
DateFormat:
```
export enum DateFormat {
  DAY = 'day',
  WEEK = 'week',
  MONTH = 'month',
  YEAR = 'year',
}
```
The same value is passed in the variable. It's like if I wrote it by hand.

**Reply by nurul3101** [reply]
Do you get the same error if you invoke Raw query like this?
```ts
this.prismaService.$queryRaw`
          select count(*)::integer, date_part(${date}, v.timestamp) as "timestamp"
          from "Visitor" v
          where (v.timestamp between ${from}::timestamp and  ${to}::timestamp) and project_tenant_id = ${project_tenant_id}
          group by date_part(${date}, v.timestamp)
          order by timestamp
      ;`
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
